### DISCUSSION TITLE
FALLMOT Entität für Ventilposition - Ein Problem oder habe ich eine Änderung verpasst :) ?

### DISCUSSION THREAD

**Discussion by kitty99hub** [SukramJ/aiohomematic#1611] [General]
Hallo,

Mir ist in HA jetzt aufgefallen, dass die Position meiner Stellantriebe der Fussbodenheizung nicht mehr angezeigt wird und HA meldet dass die Entität nicht mehr von der Integration bereitgestellt wird....
Hier eine Beispielentität für den Öffnungsgrad eines Ventils: "sensor.hmip_falmot_c12_001b9f29a2b84c_level_ch1"

Gleichzeitig habe ich jetzt aber auch noch ein paar Entitäten gesehen, die vermutlich neu dazugekommen sind und jetzt auch das setzen einer Ventilposition über HA ermöglichen.... Soweit ich mich erinnern kann, gab es diese noch nicht als ich vor ca. einem Jahr mein Fallmot installiert habe...
Dabei handelt es sich um die Entitäten vom type "NUMBER", wie z.B. diese: "number.fussbodenheizungsaktor_og1_1_bad_level"

An sich sieht das ganze nach einer tollen Verbesserung aus, bin aber irgendwie sehr darüber verwundert dass ich das irgendwie komplett verpasst und übersehen habe...., da ich immer alle changes gründlich durchlese :) :)

Bevor ich mich jetzt dran mache, das alles wieder in Ordnung zu bringen, würde ich mich darüber freuen, wenn jemanden bestätigen könnte, dass da Änderungen eingeführt wurden....., nicht dass doch noch irgendetwas anderes bei mir schief gelaufen ist.

- Die alten Entitäten vom type SENSOR der Ventilpositionen sind dann ja obsolete und können gelöscht werden und durch die neuen vom Type NUMBER ersetzt werden !?
- Die Entitäten vom Type NUMBER die das setzen der Ventilposition in HA ermöglichen kamen neu hinzu?
- Wann wurden diese Änderungen eingefügt? In welchem release könnte ich das nachlesen?

Vielen Dank
Beste Grüße

**Comment by SukramJ** [comment]
Du wirst ein FW Update beim Falmot gemacht haben.

**Reply by kitty99hub** [reply]
Hi,
ja das stimmst, so einer stand auch zur Verfügung :)
Haben diese Entitätsänderungen dann nur mit der neuen Firmware zu tun oder musste da auch gleichzeitig die Integration angepasst werden...?
Mir ist das genaue Zusammenspiel nicht wirklich gut bekannt, denke aber dass wenn der Hersteller etwas an seinem Gerät/der Firmware ändert, dies erst durch eine Anpassung der Integration auch mit in HA implementiert wird!?

Aber was heisst das jetzt konkret für die verschwundenen und neuen Entitäten.....?
Ist das so wie in meinem ersten Beitrag beschrieben und auch die weitere Vorgehensweise die richtige?

Die Firmware die ich aufgespielt habe ist auch nicht gerade ganz neu und schon einige Zeit raus..... und es ist auch aus solchen Gründe, dass ich nie direkt neue Firmware usw installiere, da ich immer mal abwarte was es an Änderungen und notwendigen Anpassungen mit sich bringt. Hätte halt gedacht, dass wenn solche Änderungen mitkommen, ich auch irgendwo etwas darüber lesen würde.
Vielleicht habe ich es aber wirklich irgendwo übersehen/überlesen....

Thx

**Reply by SukramJ** [reply]
Die Änderung kommt durch die Firmware und die Integration muss nicht geändert werden, da der Falmot automatisch erkannt wird. Einfach die alten Sensoren löschen. Number Entitäten sind wie Sensoren für Zahlen, nur das die Werte auch geschrieben werden können. Ob das auch Sinn macht kann ich dir nicht sagen.

**Reply by kitty99hub** [reply]
Ok, danke für deine schnelle Antwort.
Das Schreiben macht zB bei mir Sinn, da ich so die Ventilposition direkt aus HA heraus beeinflussen kann, unabhängig davon was das Wandthermostat vorgibt.
Das hat mich in der Vergangenheit schon gestört, da ich so zB eine Mindestventilposition (offen) festlegen wollte, auch wenn das Thermostat eigentlich auf 0% fährt.
Gründe dafür waren:
- Dass ich das komplette Abkühlen eines Fussbodens vermeiden wollte.
- Immer ein Mindestfluss gegeben ist und die bis dahin nicht geregelte Heizkreispumpe nicht gegen einen voll geschlossenen Heizkreisverteiler drückt.

Wünsche noch einen schönen Sonntag :)

**Reply by ivo7B4** [reply]
Ich habe dies auch festgestellt und die oberen Kommentare kann ich alle nachvollziehen, danke!

Weiss jemand, wenn dies überschrieben wird, für wie lange dass dies überschrieben ist? Ist ab dann die Regelung über den Wandthermostat dann deaktiviert (da der Wert überschrieben wurde)? Oder wird der neue Wert vom Wandthermostat genommen, sobald dieser einen neuen Sollwert vorgibt?

Vielen Dank!

**Comment by x4N70pHyLL** [comment]
Ich hatte das hier beschrieben: [https://github.com/danielperna84/hahomematic/issues/1676](url)

Ich hatte KEIN Firmwareupdate der FALMOT gemacht! Lediglich die Integration in Home Assistant geupdated vor einigen Tagen.

Alles was du schreibst kann ich so bestätigen, ich beobachte bisher ein instabileres Verhalten (Thermostate springen immer mal wieder auf Heizen statt auf Kühlen zu bleiben; Ventile bleiben auf ihrer Position stehen).

Ein Ventil war bei mir sehr hartnäckig, ein Trennen der FALMOT vom Strom hat dann geholfen, seitdem läuft es wieder auf "Automatik".

Wann die Ventile wenn sie von Hand gesteuert wurden wieder in Automatikbetrieb zurückgehen habe ich auch noch nicht nachvollziehbar herausgefunden.

Eigentlich ein Drama, das sowas "einfach mal" gemacht wird und nirgends halbwegs auffindbar dokumentiert ist.

Langsam wundere ich mich bei dem Verhalten vom Hersteller auch nicht mehr über die Einstellung hier bei der "nachgelagerten" Software wie Raspberrymatic oder HA-Integration. Dort hatte ich schon mal Bugs gemeldet. "Kommt vom Hersteller, daher mir egal, closed" - aus anderen Projekten kenne ich es so, dass es dann an das übergeordnete / vorgelagerte Projekt weitergegeben wird und das Ticket mit dem dortigen zusammen getrackt wird.

Hier scheint es eher so zu laufen "Liegt beim Hersteller, dann weg"

**Reply by SukramJ** [reply]
> Ich hatte das hier beschrieben: [https://github.com/danielperna84/hahomematic/issues/1676](https://github.com/danielperna84/hahomematic/discussions/url)

Könnte auch passiert sein weil mit der 1.65.0 der Cache gelöscht wurde, und Monate vorher ein FW Update eingespielt wurde.

> Eigentlich ein Drama, das sowas "einfach mal" gemacht wird und nirgends halbwegs auffindbar dokumentiert ist.

Hier wurde nicht dokumentiert, weil hier nichts geändert wurde! Ich dachte das hatte ich halbwegs verständlich geschrieben.

> "Kommt vom Hersteller, daher mir egal, closed" - aus anderen Projekten kenne ich es so, dass es dann an das übergeordnete / vorgelagerte Projekt weitergegeben wird und das Ticket mit dem dortigen zusammen getrackt wird.
> Hier scheint es eher so zu laufen "Liegt beim Hersteller, dann weg"

Da hast Du recht. 
Meinst du ich habe nichts besseres zu tun als mich auch noch um jeden vermeintlichen "Mist" zu kümmern, der von anderen produziert wird?
Ist deine Erwartung, das ich mich um dein verkorkstes Gerät kümmere und ein Ticket bei eq3 oder RaspberryMatic aufmache?
Ich habe keinen FALMOT und somit habe ich auch kein Problem damit. 
Ein Wert wird in HA geändert und auch so in der CCU angezeigt. Fertig.
Hier werden Geräte integriert. Es werden Datenpunkte gelesen und geschrieben, und das muss möglichst zuverlässig funktionieren.

**Reply by x4N70pHyLL** [reply]
> 
> Könnte auch passiert sein weil mit der 1.65.0 der Cache gelöscht wurde, und Monate vorher ein FW Update eingespielt wurde.
> 
Das wäre ja mal ein Ansatz (siehe unten)
> 
> Hier wurde nicht dokumentiert, weil hier nichts geändert wurde! Ich dachte das hatte ich halbwegs verständlich geschrieben.
> 
Das hatte ich verstanden und war auch nicht meine Erwartung. Wenn es vom Cache löschen kommt dann gibt es einen Grund, warum es sich eben jetzt geändert hat. Ansonsten muss ja an anderer Stelle (wie z.B. Firmware oder CCU) was gedreht worden sein und bei so einer Änderung hätte ich erwartet, dass die nicht "einfach mal so" passiert. Ich versuche ja schon, alle Changelogs so weit zu lesen, dass mir das über den Wege gelaufen sein müsste, nicht nur die von der Integration hier.


> Da hast Du recht. Meinst du ich habe nichts besseres zu tun als mich auch noch um jeden vermeintlichen "Mist" zu kümmern, der von anderen produziert wird? Ist deine Erwartung, das ich mich um dein verkorkstes Gerät kümmere und ein Ticket bei eq3 oder RaspberryMatic aufmache? Ich habe keinen FALMOT und somit habe ich auch kein Problem damit. Ein Wert wird in HA geändert und auch so in der CCU angezeigt. Fertig. Hier werden Geräte integriert. Es werden Datenpunkte gelesen und geschrieben, und das muss möglichst zuverlässig funktionieren.

Habe ich die Erwartung, dass Du Dich um mein Gerät kümmerst -> Eindeutig nein.

Um jeden Mist -> auch nein.

Um Tickets bei Raspberrymatic / eq3 -> Erwartung nein, Hoffnung schon. Das was beim Nutzer ankommt ist mMn das, was zählt. Wann das nicht passt, haben alle in der Kette ein Problem, daher ist meine Meinung (und auch mein Erleben bei anderen tollen Projekten), dass Abhängigkeiten in anderen Projekten mit betrachtet / Issues getreckt werden.

Dass Du das anders siehst ist ja völlig in Ordnung. Daher sagte ich ja oben, dass es Sich um meinen Eindruck handelt, wie hier in diesem und Nachbarprojekten damit umgegangen wird. Du hast das dann ja bestätigt, dass es es so ist.

**Reply by SukramJ** [reply]
Ich werde keine substanzlosen Tickets bei Raspberrymatic / eq3 eröffnen, wenn ich die Probleme mangels eigenem Gerät nicht nachvollziehen kann, und die Informationen nur auf HörenSagen bestehen. Ich kann mir nicht vorstellen das "Nachbarprojekte" (welche das auch immer sind) das anders handhaben.

Wass hält dich davon ab selber ein Ticket aufzumachen? Die Referenz dazu könntest Du natürlich hier verlinken, und vielleicht beteiligen sich auch andere Betroffene. Nachfragen hab ich bisher immer beantwortet.

> Das was beim Nutzer ankommt ist mMn das, was zählt. Wann das nicht passt, 

Es gibt hier Leute die arbeiten mit, haben konstruktive Vorschlage oder arbeiten aktiv an den eigenen Tickets/Themen mit.
Es gibt aber auch ein einige, die erwarten, das andere für sie die Arbeit machen, die sich z.Bsp. über die Dokumentation beschweren, aber sich selber nicht einbringen, oder auch nicht auf Rückfragen reagieren.
Du kannst dir überlegen zu welche Gruppe Du gehörst bzw. gehören möchtest.

**Comment by ivo7B4** [comment]
@SukramJ , ich möchte mich bei Dir/Euch für die super Arbeit bedanken, welche ihr hier macht. Dass sich die Entität ändert, wenn sich dies beim FAMLOT ändert, ist für mich gut nachvollziehbar, da dies nach meinem Verständnis genau eine der wesentlichen Ideen dieser neueren Integration ist, dass die Geräte und die entsprechenden Identitäten direkt erkannt werden und neue Geräte damit automatisch und vollständig direkt unterstützt werden.

Bisher habe ich die Raumthermostaten auf off und dann wieder auf heating/cooling geschaltet, sofern ich versehentlich auf die Position gekommen bin und da ich nicht wusste, ob und ab wann der Wert dann wieder automatisch gesetzt wird. Hier ein know-how-Austausch von den Erfahrungen würde mich interessieren. Ich weiss nicht, ob ich bei eq3 keine gute Antwort erhalten würde, wenn ich dies mit einem "nicht eq3 Gerät" beschreibe. Eventuell versuche ich dies mal noch.

@x4N70pHyLL : Die weiteren Probleme, welche Du seither beschrieben hast, kann ich nicht nachvollziehen. Umschalten Heizen und Kühlen z.B. funktioniert bei mir z.B. einwandfrei. 
Und noch eine Info: Ich glaube, die Entitäten werden nur bei einem reload der Integration neu geladen, oder bei einem Neustart (eventuell sogar nur bei vollständigem reboot) von Home Assistant. Bei mir war zwar vorher ein firmware update, aber es hat dann noch einige Zeit gedauert, bis die Entitäten gewechselt hatten. Ich gehe ebenfalls davon aus, wie von @SukramJ beschrieben, dass dies aufgrund des FW updates des FALMOTs war. Wahrscheinlich war bei mir der nächste reboot erst einiges später (und interessanterweise hatte ich dies über mehrere Tage, zuerst die Änderung bei den einen FALMOTs, dann bei den anderen). Eventuell ist dies bei Dir auch so, nur der firmware update eventuell bereits etwas länger seither?

Und eventuell für alle interessant, die dieses Problem auch haben und ein versehentliches Verstellen der Entitäten für diese Position verhindern wollen: Dies lässt sich im YAML mode der entity card mit hinzufügen des types simple-entity z.B. wie folgt verhindern:
  - entity: BezeichnungDerEntität_Fussbodenheizung_verknupfungspartner_raumthermostat_level
    type: simple-entity

Ist nicht perfekt, da nach einem Draufklicken eine Verstellung immer noch möglich ist (muss dann aber auch noch verstellt und bestätigt werden). Damit aber wird dieser nicht mehr einfach versehentlich verstellt, nur weil auf Pfeil hoch oder runter gedrückt wurde (wie dies ohne simple-entity bei dieser Input Number Entität sonst leider sofort passiert; was aber eine allgemein Bedienung von HomeAssistant von input numbers ist und keinen Zusammenhang mit dieser Integration hat).

Ich hoffe, mein Kommentar unterstützt und bewirkt keine zusätzliche Verwirrung.

**Reply by x4N70pHyLL** [reply]
@ivo7B4 Mir hat das geholfen, danke!

DAs Umschalten Heizen/Kühlen scheint einmalig gewesen bzw. bis es sich "wieder eingerüttelt" hat, mir sind die Thermostate ziemlich durcheinander gekommen, inzwischen bleiben sie wieder stabil auf "Kühlen" stehen.
Das Auto/Aus-Schalten der Thermostate hilft in der Regel bei mir auch, um die Ventilposition wieder automatisch zu finden.
In den "Sonderfällen" wo es nicht geholfen hat, musste ich dann die FALMOT Neustarten.
Zum Firmware-Update: Ich habe die FALMOTs seit März im Einsatz. Bei Inbetriebnahme habe ich das aktuelle Firmware-Update gemacht (auf 1.4.10), seitdem habe ich weder eines angezeigt bekommen, noch eines gemacht. 
 Seitdem habe ich definitiv neu gestartet etc.

Weiß jemand, für was die "Bediensperre" ist? Die ist bei mir ebenfalls neu dazugekommen.

**Comment by alexirion** [comment]
Ich bin mit der Falmot C12 auch auf Firmware 1.4.10, aber kann die Ventilstellung nicht per XML-API verändern.

http://192.168.178.98/addons/xmlapi/statechange.cgi?ise_id=3061&new_value=0.75

Es kommt keine Fehlermeldung, aber der Level Wert wird wird weiterhin als "" angezeigt:

`<channel name="HmIP-FALMOT-C12 001B9F29A2B41E:1" ise_id="3052">
<datapoint name="HmIP-RF.001B9F29A2B41E:1.DEW_POINT_ALARM" type="DEW_POINT_ALARM" ise_id="3053" value="false" valuetype="2" valueunit="" timestamp="1745422607"/>
<datapoint name="HmIP-RF.001B9F29A2B41E:1.EMERGENCY_OPERATION" type="EMERGENCY_OPERATION" ise_id="3054" value="false" valuetype="2" valueunit="" timestamp="1745422607"/>
<datapoint name="HmIP-RF.001B9F29A2B41E:1.EXTERNAL_CLOCK" type="EXTERNAL_CLOCK" ise_id="3058" value="false" valuetype="2" valueunit="" timestamp="1745422607"/>
<datapoint name="HmIP-RF.001B9F29A2B41E:1.FROST_PROTECTION" type="FROST_PROTECTION" ise_id="3059" value="false" valuetype="2" valueunit="" timestamp="1745422607"/>
<datapoint name="HmIP-RF.001B9F29A2B41E:1.HUMIDITY_LIMITER" type="HUMIDITY_LIMITER" ise_id="3060" value="false" valuetype="2" valueunit="" timestamp="1745422607"/>
<datapoint name="HmIP-RF.001B9F29A2B41E:1.LEVEL" type="LEVEL" ise_id="3061" value="" valuetype="4" valueunit="100%" timestamp="1744971029"/>
<datapoint name="HmIP-RF.001B9F29A2B41E:1.LEVEL_STATUS" type="LEVEL_STATUS" ise_id="3062" value="1" valuetype="16" valueunit="" timestamp="1745422515"/>
<datapoint name="HmIP-RF.001B9F29A2B41E:1.VALVE_STATE" type="VALVE_STATE" ise_id="3063" value="7" valuetype="16" valueunit="" timestamp="1745422607"/>
</channel>`

Sollte das nun mit der aktuellen Firmware gehen, oder nicht? Oder muss ich evtl. auch die Debmatic updaten?

**Comment by Elijen** [comment]
Does writing to valve position in HA actually work and changes valve position on the device?

@x4N70pHyLL Have you managed to find a way to resume automatic operation? I would like to control the valves manually in some scenarios, but need it to work automatically most of the time.

**Reply by x4N70pHyLL** [reply]
> @x4N70pHyLL Have you managed to find a way to resume automatic operation? I would like to control the valves manually in some scenarios, but need it to work automatically most of the time.

You just have to test it, most of the times it works, I did the manual setting only once and all but one or two resumed I think. Maybe set the Hetaing Group to off and auto again, that could help
